Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs

```diff
@@ -162,16 +147,18 @@ public async Task AddOrUpdateCalendarAsync(Calendar calendar)
                 calendar.ExternalID = deviceCalendar.CalendarIdentifier;
 
                 // Update color in case iOS assigned one
-                if (deviceCalendar?.CGColor != null)
+                if (deviceCalendar.CGColor != null)
+                {
                     calendar.Color = ColorConversion.ToHexColor(deviceCalendar.CGColor);
+                }
             }
             else
             {
                 deviceCalendar.Title = calendar.Name;
 
-                if (!string.IsNullOrEmpty(calendar.Color))
+                if (calendar.Color != null && !string.IsNullOrEmpty(calendar.Color))
                 {
-                    deviceCalendar.CGColor = ColorConversion.ToCGColor(calendar.Color);
+                    deviceCalendar.CGColor = ColorConversion.ToCGColor(calendar.Color) ?? throw new InvalidOperationException("Invalid color");
```

Comment:
This seems like it should be an `ArgumentException`, not `InvalidOperationException`. The color was passed-in via the `calendar` argument.

Reply:
Fixed